service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/dataset/CopyBigQueryDatasetDifferentRegionStep.java

```diff
@@ -0,0 +1,155 @@
+    this.gcpCloudContextService = gcpCloudContextService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    FlightMap inputParameters = flightContext.getInputParameters();
+    FlightUtils.validateRequiredEntries(
+        inputParameters, ControlledResourceKeys.DESTINATION_DATASET_NAME);
+
+    UUID destinationWorkspaceId =
+        inputParameters.get(ControlledResourceKeys.DESTINATION_WORKSPACE_ID, UUID.class);
+
+    String destinationProjectId =
+        gcpCloudContextService.getRequiredGcpProject(destinationWorkspaceId);
+
+    String destinationDatasetId =
+        inputParameters.get(ControlledResourceKeys.DESTINATION_DATASET_NAME, String.class);
+
+    Map<String, Value> params = new HashMap<>();
+    String sourceDatasetId = sourceDataset.getDatasetName();
+    String sourceProjectId = sourceDataset.getProjectId();
+    params.put("source_project_id", Value.newBuilder().setStringValue(sourceProjectId).build());
+    params.put("source_dataset_id", Value.newBuilder().setStringValue(sourceDatasetId).build());
+
+    TransferConfig transferConfig =
+        TransferConfig.newBuilder()
+            .setDestinationDatasetId(destinationDatasetId)
+            .setDisplayName(destinationDatasetId)
+            .setScheduleOptions(ScheduleOptions.newBuilder().setDisableAutoScheduling(true))
+            .setDataSourceId("cross_region_copy")
+            .setParams(Struct.newBuilder().putAllFields(params).build())
+            .build();
+
+    try (DataTransferServiceClient dataTransferServiceClient = DataTransferServiceClient.create()) {
+      ProjectName parent = ProjectName.of(destinationProjectId);
+      String petSaEmail =
+          SamRethrow.onInterrupted(
+              () ->
+                  samService.getOrCreatePetSaEmail(
+                      gcpCloudContextService.getRequiredGcpProject(destinationWorkspaceId),
+                      userRequest.getRequiredToken()),
+              "enablePet");
+
+      TransferConfig config =
+          dataTransferServiceClient.createTransferConfig(
+              CreateTransferConfigRequest.newBuilder()
+                  .setParent(parent.toString())
+                  .setTransferConfig(transferConfig)
+                  .setServiceAccountName(petSaEmail)
+                  .build());
+
+      var now = Instant.now();
+
+      dataTransferServiceClient.startManualTransferRuns(
+          StartManualTransferRunsRequest.newBuilder()
+              .setParent(config.getName())
+              .setRequestedRunTime(
+                  com.google.protobuf.Timestamp.newBuilder()
+                      .setSeconds(now.getEpochSecond())
+                      .setNanos(now.getNano()))
+              .build());
+
+      DataTransferServiceClient.ListTransferRunsPagedResponse runs =
+          dataTransferServiceClient.listTransferRuns(config.getName());
+
+      for (TransferRun run : runs.iterateAll()) {
```

Comment:
a lot of this is common with the other BQ clone step - can you refactor this into a common retry method, or use one of the retry utils if they fit?

Reply:
Yes! There's a new `getWithRetry` function now being used.